manual/manual_compiler.rst

```diff
@@ -817,3 +817,198 @@ This measurement partitioning is built into the :py:meth:`get_operator_expectati
+are observationally indistinguishable (on an ideal device), and reduce noise by
+eliminating unnecessary operations from the beginning or end of the circuit.
+
+First of all, tket provides methods to `annotate` a qubit (or all qubits) as
+being initialized to zero, or discarded at the end of the circuit, or both.
+
+.. jupyter-execute::
+
+    from pytket import Circuit
+
+    c = Circuit(2)
+    c.Y(0)
+    c.CX(0,1)
+    c.H(0)
+    c.H(1)
+    c.Rz(0.125, 1)
+    c.measure_all()
+    c.qubit_create_all()
+    c.qubit_discard_all()
+
+The last two lines tell the compiler that all qubits are to be initialized to
+zero and discarded at the end. The methods :py:meth:`Circuit.qubit_create` and
+:py:meth:`Circuit.qubit_discard` can be used to achieve the same on individual
+qubits.
+
+Note that we are now restricted in how we can compose our circuit with other
+circuits. When composing after another circuit, a "created" qubit becomes a
+Reset operation. Whem composing before another circuit, a "discarded" qubit may
+not be joined to another qubit unless that qubit has itself been "created" (so
+that the discarded state gets reset to zero).
+
+Initial simplification
+~~~~~~~~~~~~~~~~~~~~~~
+
+When the above circuit is run from an all-zero state, the Y and CX gates at the
+beginning just have the effect of putting both qubits in the "1" state (ignoring
+unobservable global phase), so they could be replaced with two X gates. This is
+exactly what the :py:meth:`SimplifyInitial` pass does.
+
+.. jupyter-execute::
+
+    from pytket.passes import SimplifyInitial
+
+    SimplifyInitial().apply(c)
+    print(c.get_commands())
+
+This pass tracks the state of qubits known to be initialised to zero forward
+through the circuit, for as long as the qubits remain in a computational basis
+state, either removing gates (when they don't change the state) or replacing
+them with X gates (when they invert the state).
+
+By default, this pass also replaces Measure operations acting on qubits with a
+known state by classical set-bits operations on the target bits:
+
+.. jupyter-execute::
+
+    c = Circuit(1).X(0).measure_all()
+    c.qubit_create_all()
+    SimplifyInitial().apply(c)
+    print(c.get_commands())
+
+The measurement has disappeared, replaced with a classical operation on its
+target bit. To disable this behaviour, pass the ``allow_classical=False``
+argument to :py:meth:`SimplifyInitial` when constructing the pass.
+
+Note that :py:meth:`SimplifyInitial` does not automatically cancel successive
+pairs of X gates introduced by the simplification. It is a good idea to follow
+it with a :py:meth:`RemoveRedundancies` pass in order to perform these
+cancellations.
+
+Removal of discarded operations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+An operation that has no quantum or classical output in its causal future has no
+effect and can be removed from the circuit. By marking a qubit as discarded, we
```

Comment:
Not quite "no effect", but "no *observable* effect on an ideal system". It can have effects from crosstalk and other non-local noise if there are other quantum operations still being performed elsewhere. I guess that this is stated in the intro to this section that we are considering just observables on an ideal system, so don't know if the reiteration is needed

Reply:
Done.